src/api/app/controllers/person_controller.rb

```diff
@@ -36,12 +36,12 @@ def command
   def get_userinfo
     user = User.find_by_login!(params[:login])
 
-    if user.login != User.current.login
+    if user != User.session
       logger.debug "Generating for user from parameter #{user.login}"
       render xml: user.render_axml(User.admin_session?)
     else
-      logger.debug "Generating user info for logged in user #{User.current.login}"
-      render xml: User.current.render_axml(true)
+      logger.debug "Generating user info for logged in user #{User.session.login}"
```

Comment:
This should be `User.session!.login`, right?

Reply:
I actually changed it to session because the if is already making sure there is a session, but in general you are right